Tests/SonarScanner.MSBuild.Tasks.IntegrationTests/TargetsTests/RazorTargetTests.cs

```diff
@@ -0,0 +1,160 @@
+﻿/*
+ * SonarScanner for MSBuild
+ * Copyright (C) 2016-2021 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Globalization;
+using System.IO;
+using FluentAssertions;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using SonarScanner.MSBuild.Common;
+using SonarScanner.MSBuild.Tasks;
+using SonarScanner.MSBuild.Tasks.IntegrationTests;
+using TestUtilities;
+
+namespace SonarScanner.Integration.Tasks.IntegrationTests.TargetsTests
+{
+    [TestClass]
+    public class RazorTargetTests
+    {
+        private const string ErrorLogFilePattern = "{0}.RoslynCA.json";
+
+        public TestContext TestContext { get; set; }
+
+        [TestMethod]
+        public void Razor_ForProductProject_CheckErrorLogProperties()
+        {
+            // Arrange
+            var rootInputFolder = TestUtils.CreateTestSpecificFolderWithSubPaths(TestContext, "Inputs");
+
+            var projectSnippet = $@"
+<PropertyGroup>
+<ErrorLog>%%ErrorLogFile%%</ErrorLog>
+<SonarCompileErrorLog>%%ErrorLogFile%%</SonarCompileErrorLog>
```

Comment:
Can you please explain to me how this works? I see you set this explicitly in a project file here. So it should be set even without the targets file. 

How does the targets file product gets validated then?